modules/tinymce/src/plugins/fullscreen/test/ts/browser/FullScreenPluginTest.ts

```diff
@@ -1,17 +1,38 @@
-import { Assertions, Chain, Guard, Log, Pipeline, UiFinder } from '@ephox/agar';
+import { Assertions, Chain, Log, NamedChain, Pipeline, UiFinder } from '@ephox/agar';
 import { UnitTest } from '@ephox/bedrock-client';
-import { Cell } from '@ephox/katamari';
-import { Editor as McEditor } from '@ephox/mcagar';
-import { Attribute, Html, SelectorFind, SugarBody, SugarElement } from '@ephox/sugar';
+import { Arr, Cell } from '@ephox/katamari';
+import { ApiChains, Editor as McEditor } from '@ephox/mcagar';
+import { Attribute, Classes, Css, Insert, Remove, Html, SelectorFind, SugarBody, SugarElement, SugarShadowDom } from '@ephox/sugar';
 import Editor from 'tinymce/core/api/Editor';
 
 import FullscreenPlugin from 'tinymce/plugins/fullscreen/Plugin';
 import LinkPlugin from 'tinymce/plugins/link/Plugin';
 import SilverTheme from 'tinymce/themes/silver/Theme';
 
-export const cWaitForDialog = (ariaLabel) =>
-  Chain.control(
-    Chain.fromChainsWith(SugarBody.body(), [
+interface Config {
+  label: string;
+  setupEditor: () => Chain<Record<string, any>, Record<string, any>>;
+  cleanupEditor: () => Chain<any, any>;
+}
+
+const getContentContainer = (editor: Editor) =>
+  SugarShadowDom.getContentContainer(SugarShadowDom.getRootNode(SugarElement.fromDom(editor.getElement())));
+
+const cCloseOnlyWindow = Chain.label(
+  'Close window',
+  Chain.op((editor: Editor) => {
+    const dialogs = () => UiFinder.findAllIn(getContentContainer(editor), '[role="dialog"]');
+    Assertions.assertEq('One window exists', 1, dialogs().length);
+    editor.windowManager.close();
+    Assertions.assertEq('No windows exist', 0, dialogs().length);
+  })
+);
+
+const cWaitForDialog = (ariaLabel: string) =>
```

Comment:
Maybe add the return type here to make it clear this needs an editor? eg `(ariaLabel: string): Chain<Editor, Editor> => `